src/main/java/io/nats/client/BaseConsumerContext.java

```diff
@@ -37,20 +42,24 @@ public interface BaseConsumerContext {
      *         such as the consumer was deleted on the server in the middle of use.
      * @throws JetStreamApiException the request had an error related to the data
      */
+    @Nullable
     Message next() throws IOException, InterruptedException, JetStreamStatusCheckedException, JetStreamApiException;
 
     /**
      * Read the next message with provided max wait
      * @param maxWait duration of max wait. Cannot be less than {@value BaseConsumeOptions#MIN_EXPIRES_MILLS} milliseconds.
+     *                If null, {@value BaseConsumeOptions#DEFAULT_EXPIRES_IN_MILLIS} will be used.
      * @return the next message or null if the max wait expires
      * @throws IOException covers various communication issues with the NATS
      *         server, such as timeout or interruption
      * @throws InterruptedException if one is thrown, to propagate it up
      * @throws JetStreamStatusCheckedException an exception representing a status that requires attention,
      *         such as the consumer was deleted on the server in the middle of use.
      * @throws JetStreamApiException the request had an error related to the data
+     * @throws IllegalArgumentException if maxWait less than {@value BaseConsumeOptions#MIN_EXPIRES_MILLS}
      */
-    Message next(Duration maxWait) throws IOException, InterruptedException, JetStreamStatusCheckedException, JetStreamApiException;
+    @Nullable
+    Message next(@Nullable Duration maxWait) throws IOException, InterruptedException, JetStreamStatusCheckedException, JetStreamApiException;
```

Comment:
I know it's weird to have the only parameter of a method be nullable, but I've annotated based on implementation and the underlying implementations handles if the value is not present.